### PR TITLE
VSCODE-88: use mongosh parser for the shell symbols completion

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Use mongosh parser for the shell API symbols/methods completion.
- Switch to node driver v3.5.6.
- Remove `JSON.stringify` from the logger and use `util.inspect` instead.
- Async tests improvements.
- Show LSP debug in the language server output channel.

### Checklist
- [x] New tests and/or benchmarks are included
- [x] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/logging.ts [hunk 1]
```diff
@@ -16,35 +17,35 @@ class Logger implements ILogger {
...
-      `${message} ${optionalParams ? JSON.stringify(optionalParams) : ''}`
+      `${message} ${optionalParams ? util.inspect(optionalParams) : ''}`
     );
   }
 
   public debug(message?: any, ...optionalParams: any[]): void {
     this.append(
       'DEBUG',
-      `${message} ${optionalParams ? JSON.stringify(optionalParams) : ''}`
+      `${message} ${optionalParams ? util.inspect(optionalParams) : ''}`
```
Comment (Anemy) [inline src/logging.ts:27]: 👍

File: src/test/suite/language/languageServerController.test.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    fakeShowErrorMessage = sandbox.stub(vscode.window, 'showErrorMessage');
+
+    await openPlayground(getDocUri('test.mongodb'));
+  });
+
+  after(() => {
+    sandbox.restore();
+  });
+
+  suite('user is connected and runs math operations in playground', () => {
```
Comment (Anemy) [inline src/test/suite/language/languageServerController.test.ts:46]: Nice full server tests.

File: package.json [hunk 3]
```diff
@@ -523,6 +523,7 @@
     "uuid": "^7.0.0",
     "vscode-languageclient": "^6.1.3",
     "vscode-languageserver": "^6.1.1",
+    "vscode-languageserver-textdocument": "^1.0.1",
```
[the remark this answers is not in the document]
Comment (alenakhineika) [reply package.json:526]: We need a document manager to create a connection in the language server.

https://github.com/mongodb-js/vscode/blob/VSCODE-9-shell-completion/src/language/server.ts#L23

**Comment by alenakhineika** [comment]
@Anemy updated the MongoDB service to return no completions if connecting or disconnecting. About having description for completion yeah would be nice. But currently, mongosh does not return this information.  We can easily add it later when mongosh will start to return more detailed completions.

**Comment by alenakhineika** [comment]
@Anemy there is no ticket about adding it to mongosh. They also need it for compass shell completion, but it wasn't planned for the nearest future.

**Pull request merged**